account/src/main/java/org/killbill/billing/account/dao/DefaultAccountDao.java

```diff
@@ -167,18 +171,59 @@ public Pagination<AccountModelDao> searchAccounts(final String searchKey, final
                                                           accountModelDao == null ? Collections.emptyIterator() : List.of(accountModelDao).iterator());
         }
 
+        final SearchQuery searchQuery;
+        if (searchKey.startsWith(SEARCH_QUERY_MARKER)) {
+            searchQuery = new SearchQuery(searchKey,
```

Comment:
In the 'new free form', we necessarily impose and `AND` ?

Reply:
Yes, I think that's the intuitive behavior. For instance, on GitHub, the `AND` is implicit too:

<img width="847" alt="image" src="https://github.com/killbill/killbill/assets/51940/db30eabe-1cab-4b36-89da-a720ff167eed">

That being said, we could add support for `OR` (just need to define the right syntax from an "api" perspective).